src/python_testing/TC_TIMESYNC_2_1.py

```diff
@@ -121,28 +124,23 @@ async def test_TC_TIMESYNC_2_1(self):
                 last_valid_until = dst.validUntil
                 if dst.validUntil is NullValue or dst.validUntil is None:
                     asserts.assert_equal(dst, dst_dut[-1], "DSTOffset list must have Null ValidUntil at the end")
-        elif self.check_pics("TIMESYNC.S.F00"):
-            asserts.assert_true(False, "DSTOffset is mandatory if the TZ (TIMESYNC.S.F00) feature is supported")
```

Comment:
Do we still need this check for the DSTOffset attribute existence since dst_dut could be an empty list or does the `await self.read_ts_attribute_expect_success(`  call fail if the attribute isn't found?

Reply:
For example, the test below for Read UTCTime has a `if utc_dut is NullValue:` which then fails the test.

Reply:
Hi @aronrosenberg ,
Thank you for your review and comments on this PR, please find my response below!

The elif statement you have commented was removed since if the self.supports_time_zone is not true in this instance then test step 7 would be skipped which appears to be what the elif statement commented on was checking in this instance, as time zone is a required feature for DSTOffset attribute to be supported on the device.

This is a good question and the test appears should fail if the DSTOffset atttribute was not found in the async await function that you have mentioned ` dst_dut = await self.read_ts_attribute_expect_success(endpoint=endpoint, attribute=attributes.DSTOffset)` as it is expecting it to be a supported attribute since time zone is a supported feature on the device to get to that point. await The function should display below output in the terminal at the point of failure if the attribute was not found:
AttributeError: type object 'Attributes' has no attribute 'DSTOffset'

Reply:
Makes sense, you can resolve this.

Reply:
Thank you!